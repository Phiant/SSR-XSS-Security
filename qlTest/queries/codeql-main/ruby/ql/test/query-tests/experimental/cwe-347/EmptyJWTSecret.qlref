experimental/cwe-347/EmptyJWTSecret.ql