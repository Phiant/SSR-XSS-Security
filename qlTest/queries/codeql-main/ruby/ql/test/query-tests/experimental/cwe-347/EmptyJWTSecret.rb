require 'jwt'

payload = { foo: 'bar' }

# BAD: the token is not signed
token1 = JWT.encode({ foo: 'bar' }, "secret", 'none')

# BAD: the secret used is empty
token2 = JWT.encode({ foo: 'bar' }, nil, 'HS256')

# BAD: the secret used is empty
token3 = JWT.encode({ foo: 'bar' }, "", 'HS256')

# GOOD: the token is signed
token4 = JWT.encode({ foo: 'bar' }, "secret", 'HS256')