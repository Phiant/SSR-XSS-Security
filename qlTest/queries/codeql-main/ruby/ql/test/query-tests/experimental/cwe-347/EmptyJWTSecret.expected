| EmptyJWTSecret.rb:9:21:9:34 | call to [] | This JWT encoding uses an empty key or none algorithm. |
| EmptyJWTSecret.rb:12:21:12:34 | call to [] | This JWT encoding uses an empty key or none algorithm. |
| MissingJWTVerification.rb:6:38:6:44 | payload | This JWT encoding uses an empty key or none algorithm. |
